### PR TITLE
Implementing useRef hook in userDetails QR 

### LINKED ISSUE #6495: [IMPROVEMENTS] Implementing useRef hook in userDetails QR
### What would you like to share?

image

I noticed that the userDetails.js component was using "document.querySelector" to target elements, which I believe is not the most convenient approach. Instead, I suggest using the "useRef" hook for this purpose. I have already made the necessary changes to the code. Please assign me ill be very happy to make a PR ....Thanks 

### Additional information

No

### PR DESCRIPTION
**Pull request by shobhitexe**
Closes #6495 

## Changes proposed

Implemented use of "useRef" hook instead of "document.getElementbyid"

## Check List (Check all the applicable boxes) 

- [x] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

## Screenshots

image

## Note to reviewers

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/6563"></a>

### CODE REVIEW

File: components/user/UserProfile.js [hunk 1]
```diff
@@ -73,7 +77,8 @@ function UserProfile({ BASE_URL, data }) {
         </div>
       )}
 
-      <div className="flex justify-center my-4">
+      {/* Passed Ref object as the ref attribute to the JSX of the DOM node of QR */}
+      <div className="flex justify-center my-4" ref={qrRef}>
         {qrShow && (
           <QRCodeCanvas
```
Comment (dan-mba) [inline components/user/UserProfile.js:83]: The id can be removed form this component as it is no longer used

Comment (shobhitexe) [reply components/user/UserProfile.js:83]: Yes Done removed

**Review by dan-mba — APPROVED** [review]
Looks good.
Thanks for the PR

**Review by dan-mba — APPROVED** [review]

**Comment by eddiejaoude** [comment]
Thank you for the contribution and great collaboration everyone 💪

**Pull request merged**